corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js

```diff
@@ -276,20 +282,23 @@ hqDefine("cloudcare/js/formplayer/utils/utils", function () {
 
         this.setQueryData = function ({ inputs, execute, forceManualSearch, initiatedBy}) {
             var selections = Utils.currentUrlToObject().selections;
+            var queryKey = sessionStorage.queryKey;
             this.queryData = this.queryData || {};
 
             const queryDataEntry = _.defaults({
                 inputs: inputs,
                 execute: execute,
                 force_manual_search: forceManualSearch,
                 selections: selections,
-            }, this.queryData[sessionStorage.queryKey]);
+            }, this.queryData[queryKey]);
 
             if (initiatedBy !== null && initiatedBy !== undefined) {
                 queryDataEntry.initiatedBy = initiatedBy;
             }
-
-            this.queryData[sessionStorage.queryKey] = queryDataEntry;
+            if (!_.isEmpty(inputs)) {
```

Comment:
should this be handled inside `setQueryInputs` and set value of inputs to `{}` if it's empty?

Otherwise, would this check also need to be done in `query.js` for 
`formplayerUtils.setQueryInputs(this.getAnswers());`

Reply:
I don't think so, but I also don't think the check I have is quite right. Sometimes `inputs` needs to be intentionally set to `null` or an empty object `{}`. I was hoping not to have to investigate every place this is called, but I think I do need to in working on an issue that's occurring now between default values and sticky query inputs.